archive_utils.py

```diff
@@ -1205,40 +1226,29 @@ def decimate_data(cleaned_archives,output_dir,cparams,logger):
             decimation_info = decimation_info.replace(np.nan, 'None', regex=True)
             decimation_info = decimation_info.values.tolist()
 
-            #If the recorded number of channels is 928 (Mostly L-band observations)
-            if len(freqs) == 928:
-
-                for num in range(0,len(decimation_info)):
-                    while 'None' in decimation_info[num]: decimation_info[num].remove('None')
-                    if decimation_info[num][0] == psrname:
-                        for item in decimation_info[num]:
-                            if not item == psrname:
-                                extension = get_extension(item,False)
-                                if not os.path.exists(os.path.join(decimated_path,"{0}.{1}".format(archive_name,extension))):
-                                    pam_command = "pam {0} -e {1} -u {2} {3}".format(item,extension,decimated_path,clean_archive)
-                                    logger.info("Producing {0} archives".format(extension))
-                                    proc_pam = shlex.split(pam_command)
-                                    subprocess.call(proc_pam)
-                                    processed_archives.append(os.path.join(decimated_path,"{0}.{1}".format(archive_name,extension)))
-                                else:
-                                    logger.info("{0}.{1} exists".format(archive_name,extension))
-                                    processed_archives.append(os.path.join(decimated_path,"{0}.{1}".format(archive_name,extension)))
-
 
+            if (rcvr == "LBAND"):
 
+                if len(freqs) == 928:
 
-            #If the recorded number of channels is greater than 928 (could be either L-band or UHF)
+                    for num in range(0,len(decimation_info)):
```

Comment:
I personally don't like this style and think it is harder to read. A better way to make the loop in python is like so:
```
for dec_info in decimation_info:
    print(dec_info)
```
It's a bit clearer what you're doing this way and you don't have to keep slapping `[num]` on the end

Reply:
Agreed - I think this is code I inherited and never got around to updating. Changes will be implemented in a new commit shortly.

Reply:
Changes implemented in latest commit.